vlingo-reactive-messaging-patterns/src/test/java/io/vlingo/reactive/messaging/patterns/competingconsumer/CompetingConsumerTest.java

```diff
@@ -0,0 +1,36 @@
+// Copyright © 2012-2018 Vaughn Vernon. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the
+// Mozilla Public License, v. 2.0. If a copy of the MPL
+// was not distributed with this file, You can obtain
+// one at https://mozilla.org/MPL/2.0/.
+package io.vlingo.reactive.messaging.patterns.competingconsumer;
+/**
+ * CompetingConsumerTest provides an example of implementing
+ * and using a vlingo Router.
+ */
+import org.junit.Test;
+
+import io.vlingo.actors.Definition;
+import io.vlingo.actors.World;
+import io.vlingo.actors.testkit.TestUntil;
+
+public class CompetingConsumerTest {
+
+  @Test
+  public void testThatConsumersCompete() {
+    final World world = World.startWithDefaults("competing-consumer-test");
+    final int poolSize = 4;
+    final int messagesToSend = 8;
+    final TestUntil until = TestUntil.happenings(messagesToSend);
+    final WorkConsumer workConsumer = world.actorFor(
+            Definition.has(WorkRouterActor.class, Definition.parameters(poolSize, until)),
+            WorkConsumer.class);
+    
+    for (int i = 0; i < messagesToSend; i++) {
+      workConsumer.consumeWork(new WorkItem(i));
```

Comment:
@DaveMuirhead You can use `TestActor::actor().viewTestState()`, where `TestActor` is the "proxy" returned from `actorOf(...)` when using `TestWorld`. The `Actor::viewTestState()` should throw `IllegalStateException` if the `World::isTestWorld()` returns `false`, with an override in `TestWorld` that returns `true`. You could provide a `Completes<Blah>` method on the actor's protocol, but if it's only used for test then it's sort of a wart. Another completely different approach would be to implement something on `vlingo-telemetry` for the `Actor` to provide some specific output that you could use for test state, either async or sync. Talk with @kmruiz about that (he may already support it).

Reply:
@DaveMuirhead I didn't merge because I was unsure of your satisfaction with the current test. If you want to try out one or more of the suggestions and have me re-review, that works.